project/project.go

```diff
@@ -65,15 +65,18 @@ func NewProject(context *Context, runtime RuntimeProject, parseOptions *config.P
 	}
 
 	if context.EnvironmentLookup == nil {
-		cwd, err := os.Getwd()
+		absPath, err := filepath.Abs(context.ComposeFiles[0])
```

Comment:
Can you check to make sure that `len(context.ComposeFiles[0]) > 0`? If not we should fall back to the existing behavior (`Path: filepath.Join(cwd, ".env")`).